core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java

```diff
@@ -307,7 +313,7 @@ void forceSorterToSpill() throws IOException {
       SpillInfo[] spills,
       File outputFile,
       @Nullable CompressionCodec compressionCodec) throws IOException {
-    assert (spills.length >= 2);
+    assert (spills.length >= 2 || !allowSpillMove);
```

Comment:
yes, good catch!  actually this should have been caught by my tests, but they were misconfigured, thanks.